#nanodet-m-416
# COCO mAP(0.5:0.95) = 0.235
#             AP_50  = 0.384
#             AP_75  = 0.242
#           AP_small = 0.082
#               AP_m = 0.240
#               AP_l = 0.375
save_dir: workspace/nanodet_m_416
model:
  arch:
    name: GFL
    backbone:
      name: ShuffleNetV2
      model_size: 1.0x
      out_stages: [2,3,4]
      activation: LeakyReLU
    fpn:
      name: PAN
      in_channels: [116, 232, 464]
      out_channels: 96
      start_level: 0
      num_outs: 3
    head:
      name: NanoDetHead
      num_classes: 9
      input_channel: 96
      feat_channels: 96
      stacked_convs: 2
      share_cls_reg: True
      octave_base_scale: 5
      scales_per_octave: 1
      strides: [8, 16, 32]
      reg_max: 7
      norm_cfg:
        type: BN
      loss:
        loss_qfl:
          name: QualityFocalLoss
          use_sigmoid: True
          beta: 2.0
          loss_weight: 1.0
        loss_dfl:
          name: DistributionFocalLoss
          loss_weight: 0.25
        loss_bbox:
          name: GIoULoss
          loss_weight: 2.0
data:
  train:
    name: coco
    img_path: /home/ucar/nanodet-0.2.0/dataset-coco/train
    ann_path: /home/ucar/nanodet-0.2.0/dataset-coco/annotations/train.json
    input_size: [416,416] #[w,h]
    keep_ratio: True
    pipeline:
      perspective: 0.0
      scale: [0.5, 1.4]
      stretch: [[1, 1], [1, 1]]
      rotation: 0
      shear: 0
      translate: 0.2
      flip: 0.5
      brightness: 0.2
      contrast: [0.6, 1.4]
      saturation: [0.5, 1.2]
      normalize: [[103.53, 116.28, 123.675], [57.375, 57.12, 58.395]]
  val:
    name: coco
    img_path: /home/ucar/nanodet-0.2.0/dataset-coco/val
    ann_path: /home/ucar/nanodet-0.2.0/dataset-coco/annotations/val.json
    input_size: [416,416] #[w,h]
    keep_ratio: True
    pipeline:
      normalize: [[103.53, 116.28, 123.675], [57.375, 57.12, 58.395]]
device:
  gpu_ids: [0]
  workers_per_gpu: 1
  batchsize_per_gpu: 8
schedule:
#  resume:
#  load_model: YOUR_MODEL_PATH
  optimizer:
    name: SGD
    lr: 0.01
    momentum: 0.9
    weight_decay: 0.0001
  warmup:
    name: linear
    steps: 50
    ratio: 0.1
  total_epochs: 28
  lr_schedule:
    name: MultiStepLR
    milestones: [240,260,275]
    gamma: 0.1
  val_intervals: 10
evaluator:
  name: CocoDetectionEvaluator
  save_key: mAP

log:
  interval: 10

class_names: ['chili', 'tomato', 'potato', 'banana', 'apple', 'watermelon',
              'cola', 'cake', 'milk']
